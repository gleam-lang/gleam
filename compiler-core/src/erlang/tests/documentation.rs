use crate::assert_erl;

#[test]
fn function_with_documentation() {
    assert_erl!(
        r#"
/// Function doc!
pub fn documented() { 1 }"#
    );
}

#[test]
fn function_with_multiline_documentation() {
    assert_erl!(
        r#"
/// Function doc!
/// Hello!!
///
pub fn documented() { 1 }"#
    );
}

#[test]
fn quotes_in_documentation_are_escaped() {
    assert_erl!(
        r#"
/// "hello"
pub fn documented() { 1 }"#
    );
}

#[test]
fn backslashes_in_documentation_are_escaped() {
    assert_erl!(
        r#"
/// \hello\
pub fn documented() { 1 }"#
    );
}

#[test]
fn single_line_module_comment() {
    assert_erl!(
        r#"
//// Hello! This is a single line module comment.

pub fn main() { 1 }"#
    );
}

#[test]
fn multi_line_module_comment() {
    assert_erl!(
        r#"
//// Hello! This is a multi-
//// line module comment.
////

pub fn main() { 1 }"#
    );
}

#[test]
fn double_quotes_are_escaped_in_module_comment() {
    assert_erl!(
        r#"
//// "quotes!"

pub fn main() { 1 }"#
    );
}

#[test]
fn backslashes_are_escaped_in_module_comment() {
    assert_erl!(
        r#"
//// \backslashes!\

pub fn main() { 1 }"#
    );
}

#[test]
fn internal_function_has_no_documentation() {
    assert_erl!(
        r#"
/// hidden!
@internal
pub fn main() { 1 }"#
    );
}
