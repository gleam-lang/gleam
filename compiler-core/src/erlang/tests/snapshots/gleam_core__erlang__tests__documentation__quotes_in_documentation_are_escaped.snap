---
source: compiler-core/src/erlang/tests/documentation.rs
expression: "\n/// \"hello\"\npub fn documented() { 1 }"
---
----- SOURCE CODE

/// "hello"
pub fn documented() { 1 }

----- COMPILED ERLANG
-module(my@mod).
-compile([no_auto_import, nowarn_unused_vars, nowarn_unused_function, nowarn_nomatch]).

-export([documented/0]).

-if(?OTP_RELEASE >= 27).
-define(MODULEDOC(Str), -moduledoc(Str)).
-define(DOC(Str), -doc(Str)).
-else.
-define(MODULEDOC(Str), -compile([])).
-define(DOC(Str), -compile([])).
-endif.

-file("project/test/my/mod.gleam", 3).
?DOC(" \"hello\"\n").
-spec documented() -> integer().
documented() ->
    1.
